src/node.cc

```diff
@@ -1546,11 +1557,29 @@ static void GetActiveRequests(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
   Local<Array> ary = Array::New(args.GetIsolate());
-  int i = 0;
+  Local<Context> ctx = env->context();
+  Local<Function> fn = env->add_properties_by_index_function();
+  static const size_t argc = 5;
+  Local<Value> argv[argc];
+  size_t i = 0;
+
+  for (auto w : *env->req_wrap_queue()) {
+    if (w->persistent().IsEmpty() == false) {
+      argv[i++ % argc] = w->object();
+      if ((i % argc) == 0) {
+        HandleScope scope(env->isolate());
+        fn->Call(ctx, ary, argc, argv).ToLocalChecked();
+        for (auto&& arg : argv) {
+          arg = Local<Value>();
+        }
+      }
+    }
+  }
 
-  for (auto w : *env->req_wrap_queue())
-    if (w->persistent().IsEmpty() == false)
-      ary->Set(i++, w->object());
+  if (i % 5 > 0) {
+    HandleScope scope(env->isolate());
+    fn->Call(ctx, ary, i % 5, argv).ToLocalChecked();
```

Comment:
don't think calculating the remainder will work. First the array index is set with `i++ % argc`, incrementing the counter so the following `i % argc` check can tell whether the JS fn should be called. so the remainder check isn't the same between the two. To get it working I think it would need to look something like:
```cpp
    if (w->persistent().IsEmpty() == false) {
      const size_t remainder = i++ % argc;
      argv[remainder] = w->object();
      if (remainder == argc - 1) {
        HandleScope scope(env->isolate());
```
thoughts?